pando-rt/src/global_ptr.cpp

```diff
@@ -226,7 +231,47 @@ void store(GlobalAddress dstGlobalAddr, std::size_t n, const void* srcNativePtr)
     auto byteData = *(byteSrc + i);
     DrvAPI::write<std::byte>(byteDst + i, byteData);
   }
+/*
+  using ChunkType = std::uint64_t;
+  using BlockType = std::uint16_t;
+  const auto chunkSize = sizeof(ChunkType);
+  const auto blockSize = sizeof(BlockType);
+  const auto numChunks = n / chunkSize;
+  const auto remainderBlocks = n % chunkSize;
+  const auto numBlocks = remainderBlocks / blockSize;
+  const auto remainderBytes = remainderBlocks % blockSize;
+
+  if (numChunks > 0) {
+    auto chunkSrc = static_cast<const ChunkType*>(srcNativePtr);
+    auto chunkDst = reinterpret_cast<std::uint64_t>(dstGlobalAddr);
+    for (std::size_t i = 0; i < numChunks; i++) {
+      auto chunkData = *(chunkSrc + i);
+      const auto offset = i * chunkSize;
+      DrvAPI::write<ChunkType>(chunkDst + offset, chunkData);
+    }
+  }
 
+  const auto blocksWritten = numChunks * chunkSize;
+  if (numBlocks > 0) {
+    auto blockSrc = static_cast<const BlockType*>(srcNativePtr) + blocksWritten;
+    auto blockDst = reinterpret_cast<std::uint64_t>(dstGlobalAddr) + blocksWritten;
+    for (std::size_t i = 0; i < numBlocks; i++) {
+      auto blockData = *(blockSrc + i);
+      const auto offset = i * blockSize;
+      DrvAPI::write<BlockType>(blockDst + offset, blockData);
+    }
+  }
+
+  const auto bytesWritten = blocksWritten + numBlocks * blockSize;
+  if (remainderBytes > 0) {
+    auto byteSrc = static_cast<const std::byte*>(srcNativePtr) + bytesWritten;
+    auto byteDst = reinterpret_cast<std::uint64_t>(dstGlobalAddr) + bytesWritten;
+    for (std::size_t i = 0; i < remainderBytes; i++) {
+      auto byteData = *(byteSrc + i);
+      DrvAPI::write<std::byte>(byteDst + i, byteData);
+    }
+  }
+*/
```

Comment:
So why were all these lines added if they are to be commented out?

Reply:
These lines are for adding a layer of message granularity. Currently anything larger than 64 bits will crash the system, but I kept these lines for future usage is someday the issue is fixed.

Reply:
ah okay add that to the comment please

Reply:
Add the comment and then merge main into this so I can approve this pr.